src/extension.ts

```diff
@@ -79,7 +80,7 @@ export function activate(context: vscode.ExtensionContext) {
     });
 
     context.subscriptions.push(vscode.workspace.onDidChangeConfiguration(cfg => {
```

Comment:
The `onDidChangeConfiguration` event now uses a new event interface (`ConfigurationChangeEvent`) which has a `affectsConfiguration` function to check for changes on specific configurations. Using that should avoid using compare with _oldValues_.  Maybe adding something like a `needsRefresh` function in `AbstractLocator`, which would worry only for its configs.